src/pveclib/vec_f128_ppc.h

```diff
@@ -951,17 +1314,352 @@ test_cmpltf128_v3d (vui128_t vfa128, vui128_t vfb128)
       result = vec_xsiexpqp (q_sig, q_exp);
  * \endcode
  *
+ * \subsubsection f128_softfloat_0_0_2_2 Convert Quad-Precision to Quadword integer
+ *
+ * Convertions between quad-precision and quadword integers is
+ * complicated by the fact that the QP significand is only 113-bits
+ * while the quadword integer magnitude can be 127/128 bits.
+ * It may not be possible to represent the quadword magnitude
+ * exactly.
+ * Conversions from quad-precision float to integer may have
+ * nonzero fractions which require rounding/truncation.
+ *
+ * For POWER9 we have the
+ * <B>VXS Scalar Convert with round to zero Quad-Precision to
+ * Signed/Unsigned Doubleword
+ * <I>(xscvqpsdz/xscvqpudz)</I></B> instructions.
+ * For POWER10 we have the
+ * <B>VXS Scalar Convert with round to zero Quad-Precision to
+ * Signed/Unsigned Quadword
+ * <I>(xscvqpsqz/xscvqpuqz)</I></B> instructions.
+ * Conversion using other rounding modes require using
+ * <B>VSX Scalar Round to Quad-Precision Integer <I>(xsrqpi)</I></B>
+ * instruction.
+ *
+ * \note The <I>xsrqpi</I> instruction
+ * allows for overriding the rounding mode as an immediate operand.
+ * So a two instruction sequence can implement any of the four
+ * <B>FPSCR<sub>RN</sub></B> rounding modes plus the fifth
+ * (Round to Nearest Away) mode specific to floating point integer
+ * instructions.
  *
- * \subsubsection f128_softfloat_0_0_2_2 Convert Quadword integer to Quad-Precision
+ * For this example we will look at Convert with round to zero
+ * Quad-Precision to Unsigned Quadword.
+ * The POWER10 operation can be implemented as a single xscvqpuqz
+ * instruction. For example:
+ * \code
+static inline vui128_t
+vec_xscvqpuqz (__binary128 f128)
+{
+  vui128_t result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
```

Comment:
why the `__GNUC__` restriction? Is this really a binutils requirement? Is this needed/valuable in sample code?

Reply:
Well this was taken from the actual code and was required because GCC 9 (and earlier) builds tend not to support binutils that do support PWR10 instructions.

I am still feeling my way through the maze. Eventually I can add a configure test. But for now just trying to get stuff working.

I think that, as an example, it is useful. Most developers don't know how the peaces fit, and don't have the time/patience to figure it out.